From 858efb53813d2d15c59b1b68dba327526b040c15 Mon Sep 17 00:00:00 2001
From: Stanley.Miao <stanley.miao@windriver.com>
Date: Thu, 3 Jun 2010 17:27:26 +0800
Subject: [PATCH] rtc: make the rtc-omap wake-up capable

RTC is the only wake-up source for TI OMAPL138. rtc-omap is hardcoded not
capable to wake-up the system. Now modify it to wake-up capable and make
L138 suspend-to-RAM works.

Signed-off-by: Stanley.Miao <stanley.miao@windriver.com>
---
 drivers/rtc/rtc-omap.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/rtc/rtc-omap.c b/drivers/rtc/rtc-omap.c
index 64d9727..2730084 100644
--- a/drivers/rtc/rtc-omap.c
+++ b/drivers/rtc/rtc-omap.c
@@ -410,7 +410,7 @@ static int __init omap_rtc_probe(struct platform_device *pdev)
 	 *    power mode.  (Some chip errata report that RTC_CTRL_SPLIT
 	 *    is write-only, and always reads as zero...)
 	 */
-	device_init_wakeup(&pdev->dev, 0);
+	device_init_wakeup(&pdev->dev, 1);
 
 	if (new_ctrl & (u8) OMAP_RTC_CTRL_SPLIT)
 		pr_info("%s: split power mode\n", pdev->name);
-- 
1.6.0.3

